packages/@lwc/engine-core/src/framework/base-bridge-element.ts

```diff
@@ -67,6 +69,50 @@ function createMethodCaller(methodName: string): (...args: any[]) => any {
     };
 }
 
+type AttributeChangedCallback = (
+    this: HTMLElement,
+    attrName: string,
+    oldValue: string,
+    newValue: string
+) => void;
+
+function createAttributeChangedCallback(
+    attributeToPropMap: Record<string, string>,
+    superAttributeChangedCallback: AttributeChangedCallback
```

Comment:
At this stage, the `superAttributeChangedCallback` might be `undefined`.

```suggestion
    superAttributeChangedCallback?: AttributeChangedCallback
```